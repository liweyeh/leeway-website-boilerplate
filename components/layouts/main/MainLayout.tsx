/**
 * Main layout
 * @author Carlos Romera <cromera@adasasistemas.com>
 * @author Liwei Yeh <liwei.yeh@adasasistemas.com>
 */

// Library
import React from 'react';

// Components
import { Dropdown } from '@components';

// Types
import { LinkType } from '@/types';

// Styles
import styles from './mainLayout.module.scss';

export const links: LinkType[] = [
	{ href: '/', isExternal: false, displayName: 'Home' },
	{ href: '/theme', isExternal: false, displayName: 'Theme' },
	{ href: '/defaultError', isExternal: false, displayName: 'Default Error Page' },
	{ href: '/testing', isExternal: false, displayName: 'Testing Ground' },
	{
		href: 'https://www.qld.gov.au/legal/privacy',
		isExternal: true,
		displayName: 'External',
	},
];

export const MainLayout = ({ children }: { children: React.ReactNode }) => {
	return (
		<div className={styles.container}>
			<div>
				<Dropdown menuOptions={links} customStyles={styles.menu}>
					<div className={styles['menu-icon']}></div>
					<div className={styles['menu-icon']}></div>
					<div className={styles['menu-icon']}></div>
				</Dropdown>
			</div>
			{children}
		</div>
	);
};

export default MainLayout;
