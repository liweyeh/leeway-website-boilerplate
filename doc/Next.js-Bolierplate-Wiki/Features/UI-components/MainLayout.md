[[_TOC_]]

# Overview

- Purpose: A layout that can be used through multiple pages
- Type: Layout
- Status: ðŸš€ **Completed**
- Code: [URL](https://dev.azure.com/ADASA-Accelerator/Website-React-Booster/_git/website-boilerplate-nextjs?path=/components/layouts/main)
- Created by: Liwei Yeh <liwei.yeh@adasasistemas.com>
- Created time: 23/08/2022

# Props

\* = required

- children<`React.ReactNode`>\*: pages that you want this layout to be in

# Usage

```jsx
function MyApp({ Component, pageProps }: AppLayoutProps) {
	return (
		<ThemeContextProvider>
			<ScreenSizeContextProvider>
				<ThemeLayout>
					<MainLayout>
						<Component {...pageProps} />
					</MainLayout>
				</ThemeLayout>
			</ScreenSizeContextProvider>
		</ThemeContextProvider>
	);
}
```
