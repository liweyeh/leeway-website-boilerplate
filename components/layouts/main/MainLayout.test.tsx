import { render } from '@testing-library/react';
import { MainLayout } from './MainLayout';
import '@testing-library/jest-dom';

jest.mock('next/config', () => () => ({
	publicRuntimeConfig: {},
	serverRuntimeConfig: {},
}));
const setup = () => {
	const { asFragment } = render(
		<MainLayout>
			<div>Main Layout:</div>
		</MainLayout>,
	);
	return { asFragment };
};

test('test snapshot', () => {
	const { asFragment } = setup();
	expect(asFragment()).toMatchSnapshot();
});
