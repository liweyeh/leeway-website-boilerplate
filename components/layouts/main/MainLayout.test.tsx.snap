// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test snapshot 1`] = `
<DocumentFragment>
  <div
    class="container"
  >
    <div>
      <div
        class="dropdown menu"
        data-cy="dropdown"
      >
        <button
          aria-label="button"
          class="button-default  dropdown-button"
          type="button"
        >
          <div
            class="menu-icon"
          />
          <div
            class="menu-icon"
          />
          <div
            class="menu-icon"
          />
        </button>
        <div
          class="dropdown-menu dropdown-menu-hidden"
        >
          <a
            aria-current="false"
            class="link"
            href="/"
          >
            <p>
              Home
            </p>
          </a>
          <a
            aria-current="false"
            class="link"
            href="/theme"
          >
            <p>
              Theme
            </p>
          </a>
          <a
            aria-current="false"
            class="link"
            href="/defaultError"
          >
            <p>
              Default Error Page
            </p>
          </a>
          <a
            aria-current="false"
            class="link"
            href="/testing"
          >
            <p>
              Testing Ground
            </p>
          </a>
          <a
            aria-current="false"
            class="link"
            rel="noreferrer"
            target="_blank"
          >
            <p>
              External
            </p>
          </a>
        </div>
      </div>
    </div>
    <div>
      Main Layout:
    </div>
  </div>
</DocumentFragment>
`;
